pkg/secrets/secrets.go

```diff
@@ -17,7 +34,56 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return ValidateAWSCredentials(secret)
 	case AzureSecretType:
 		return ValidateAzureCredentials(secret)
+	case GCPSecretType:
+		return ValidateGCPCredentials(secret)
+	case FilestoreSecretType:
+		return ValidateFileStoreCredentials(secret)
 	default:
 		return errors.Errorf("Unsupported type '%s' for secret '%s:%s'", string(secret.Type), secret.Namespace, secret.Name)
 	}
 }
+
+func getLocationSecret(secret *corev1.Secret) (reposerver.Secret, error) {
+	var locationType []byte
+	var ok bool
+	if secret == nil {
+		return nil, errors.New("Secret is Nil")
+	}
+
+	if locationType, ok = (secret.Data[reposerver.TypeKey]); !ok {
+		return nil, errors.Wrapf(reposerver.ErrValidate, reposerver.MissingRequiredFieldErrorMsg, reposerver.TypeKey, secret.Namespace, secret.Name)
```

Comment:
Same here, we are validating a secret, using errors for reposerver doesn't sound correct. From the maintenance pov, if we change the format for reposerver msg to include reposerver-specific information, it won't be applicable to the methods where these vars are used. 
I don't have a strong opinion, but it looks cleaner if declare and use error msgs from the same package.